types/coin.go

```diff
@@ -51,6 +49,28 @@ func (coin Coin) String() string {
 	return fmt.Sprintf("%v%v", coin.Amount, coin.Denom)
 }
 
+// Validate returns an error if the Coin has a nonpositive amount or if
+// the Denom is invalid.
+func Validate(denom string, amount Int) error {
+	if err := validateDenom(denom); err != nil {
+		return err
+	}
+
+	if amount.LTE(ZeroInt()) {
```

Comment:
@colin-axner lets go with `LT` for now so we can get the build passing and as this was the original behavior.